variables.tf

```diff
@@ -69,6 +69,24 @@ variable "cpu" {
   type        = number
 }
 
+variable "code_pipeline_artifact_bucket" {
+  default     = ""
+  description = "Use an existing bucket for codepipeline artifacts that can be reused for multiple services."
+  type        = string
+}
+
+variable "code_pipeline_role" {
+  default     = ""
+  description = "Use an existing role for codepipeline permissions that can be reused for multiple services."
+  type        = string
+}
+
+variable "code_build_role" {
+  default     = ""
+  description = "Use an existing role for codebuild permissions that can be reused for multiple services."
```

Comment:
Maybe add, that we'll create a role otherwise? Not sure if we'll need to document the requirements for a provided role (other than the spec of the one we'll create internally)...